app/src/main/java/nl/mpcjanssen/simpletask/AddTaskBackground.kt

```diff
@@ -107,30 +108,18 @@ class AddTaskBackground : Activity() {
         val todoList = TodoList
         log.debug(TAG, "Adding background tasks to todolist {} " + todoList)
 
-        val items = ArrayList<Task>()
-        for (taskText in sharedText.split("\r\n|\r|\n".toRegex()).dropLastWhile(String::isEmpty).toTypedArray()) {
-            if (taskText.trim({ it <= ' ' }).isEmpty()) {
-                continue
-            }
-            var text: String
-            if (!appendText.isEmpty()) {
-                text = taskText + " " + appendText
-            } else {
-                text = taskText
-            }
-            val t: Task
-            if (Config.hasPrependDate) {
-                t = Task(text, DateTime.today(TimeZone.getDefault()).format(Constants.DATE_FORMAT))
-            } else {
-                t = Task(text)
-            }
-            items.add(t)
+        val lines = sharedText.split("\r\n|\r|\n".toRegex()).filterNot(String::isBlank)
```

Comment:
Here's the other change that shouldn't but might have an impact. Instead of dropping blank lines from the end and then also dropping empty lines in the middle, we just filter out all *blank* (whitespace-only) lines.